src/Microsoft.AspNetCore.DataProtection/DataProtectionServiceCollectionExtensions.cs

```diff
@@ -88,10 +87,9 @@ private static void AddDataProtectionServices(IServiceCollection services)
             {
                 var dpOptions = s.GetRequiredService<IOptions<DataProtectionOptions>>();
                 var keyRingProvider = s.GetRequiredService<IKeyRingProvider>();
-                var loggerFactory = s.GetRequiredService<ILoggerFactory>();
+                var loggerFactory = s.GetService<ILoggerFactory>() ?? NullLoggerFactory.Instance;
 
-                IDataProtectionProvider dataProtectionProvider = null;
-                dataProtectionProvider = new KeyRingBasedDataProtectionProvider(keyRingProvider, loggerFactory);
+                IDataProtectionProvider dataProtectionProvider = new KeyRingBasedDataProtectionProvider(keyRingProvider, loggerFactory);
```

Comment:
Explicit typing required. Implicit typing would cause a compiler error due to the assignment that follows.